flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -184,7 +206,64 @@ public void shutdown(Time timeout) {
 		return submitRequest(targetAddress, targetPort, httpRequest, responseType);
 	}
 
-	private <P extends ResponseBody> CompletableFuture<P> submitRequest(String targetAddress, int targetPort, FullHttpRequest httpRequest, JavaType responseType) {
+	private static Request createRequest(String targetAddress, int targetPort, String targetUrl, HttpMethodWrapper httpMethod, ByteBuf jsonPayload, Collection<FileUpload> fileUploads) throws IOException {
```

Comment:
Could we combine `targetAddress` and `targetPort` into `targetAddressPort = targetAddress + ':' + targetPort`?

Reply:
We could directly pass in `httpMethod.getNettyHttpMethod()` instead of the wrapper.